src/pages/user-profile/UserProfile.tsx

```diff
@@ -95,19 +99,32 @@ const UserProfile = () => {
           }
         />
       )}
-      {response.professionalBlock && (
-        <AboutTutorBlock data={response.professionalBlock} />
+      {userResponse?.professionalBlock && (
+        <AboutTutorBlock data={userResponse?.professionalBlock} />
+      )}
+      {userResponse?.aboutStudent && (
+        <AboutStudentBlock data={userResponse?.aboutStudent} />
       )}
-      {response.aboutStudent && (
-        <AboutStudentBlock data={response.aboutStudent} />
+      {shouldShowPresentation && (
+        <VideoPresentation
+          video={
+            userResponse?.videoLink?.[preferredRole as keyof DataByRole<string>]
```

Comment:
I don't remember how that enum is called, but use that instead of `keyof`.
```suggestion
            userResponse?.videoLink?.[preferredRole as UserRole.Student | UserRola.Tutor]
```